tests/epics/conftest.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+from bluesky.run_engine import RunEngine
+
+from ophyd_async.core._detector import StandardDetector
+from ophyd_async.core._device import DeviceCollector
+from ophyd_async.core._mock_signal_utils import set_mock_value
+
+
+@pytest.fixture
+async def ad_standard_det_factory(
+    RE: RunEngine,
+    static_path_provider,
+) -> StandardDetector:
+    async def generate_ad_standard_det(ad_standard_detector_class, number=1):
+        detector_name = ad_standard_detector_class.__name__
+        if detector_name.endswith("Detector"):
+            detector_name = detector_name[:-8]
```

Comment:
Is it because of the `len("Detector") == 8`?

Reply:
Yep, I can maybe change it to `detector_name[:-(len("Detector"))] to make it more obvious what it is doing